### ISSUE TITLE
Open Interest (Statistic Improvements)

### ISSUE THREAD

**Issue by CCefman** [AtasPlatform/Indicators#4]
Hi hope you are well. @ErtyHackward has suggested that I should open a new issue regarding this as it wasn't directly related to the issue Sebastian has previously opened.

First of all I've never tried to write a script myself so I do apologise if I don't make the most sense or if I'm asking for something which is way too much, but this would be really really useful!

I've noticed that currently there are very limited statistics that you can see on Open Interest. I was hoping you'd be able to add in OI Delta statistics so you could see it above/under every candle. Check Attachment 1 to get a better idea of what I mean. 

![OI Example 1](https://user-images.githubusercontent.com/92552626/141493566-7220df37-c217-43d6-a50d-efeff0d331d6.png)

Further Improving to this would be adding some form of ZigZag Statistics so you can see the % change of the move rather then on each candle. Ideally if you could add "Minimum ZigZag Leg %" so we can adjust that ourselves that would be incredible. Have a look at Attachment 2 to get a better idea.

![OI Example 2](https://user-images.githubusercontent.com/92552626/141494105-d84a030d-cc03-452a-8d71-95327e70fdec.png)
 
Again I apologise if this is way too much to ask or if I haven’t provided sufficient information but as open interest is incredibly important in trading this would be a very useful improvement. If you need references where I’ve got this from then it’s exocharts. Thanks a lot again, and ill keep checking the develop branch.

_Originally posted by @CCefman in https://github.com/AtasPlatform/Indicators/issues/2#issuecomment-967217592_

**Comment by Sebastian7700** [comment]
Suggestion for the coding team:

A ZigZag Indicator is already available.
Instead of adding the hole ZigZag code to OI, you could add the possibility inside of the ZigZag Indicator to change the panel on which the ZigZag should be calculated.
This also has the advantage, that one can easily add ZigZag to other indicators this way, without changing the code of these indicators.
E.g. if someone wants to use it for CVD or OBV or my two new indicator suggestions "Net Longs" and "Net Shorts", this way it is no problem.

**Comment by CCefman** [comment]
I also agree with Sebastian’s suggestions. This would be a lot more convenient as you’d have the option to add this indicator to others. The main indicators I’d be using this on would be Open Interest, Net Longs and Net Shorts (New Sebastian suggested indicators). I’m guessing this way it would also save time coding as you wouldn’t have to code ZigZag indicator to the other indicators but rather just coding the ZigZag indicator itself. 

Thanks a lot for all the improvements made so far, hoping to see this one up and running soon.
